build_tools/cmake/iree_bytecode_module.cmake

```diff
@@ -109,6 +109,20 @@ function(iree_bytecode_module)
     list(APPEND _OUTPUT_FILES "${_RULE_STATIC_LIB_PATH}" "${_STATIC_HDR_PATH}")
   endif()
 
+  if(ANDROID AND NOT _RULE_FLAGS MATCHES "iree-llvm-target-triple")
+    # Android's CMake toolchain defines some variables that we can use to infer
+    # the appropriate target triple from the configured settings:
+    # https://developer.android.com/ndk/guides/cmake#android_platform
+    #
+    # In typical CMake fashion, the various strings are pretty fuzzy and can
+    # have multiple values like "latest", "android-25"/"25"/"android-N-MR1".
+    #
+    # From looking at the toolchain file, ANDROID_PLATFORM_LEVEL seems like it
+    # should pretty consistently be just a number we can use for target triple.
+    set(_TARGET_TRIPLE "aarch64-none-linux-android${ANDROID_PLATFORM_LEVEL}")
+    list(APPEND _ARGS "--iree-llvm-target-triple=${_TARGET_TRIPLE}")
+  endif()
```

Comment:
Ehhhh... this should be reconciled with https://github.com/iree-org/iree/blob/8ebe002ebebf9e2f867cc808d8a11e8658ea9ea7/build_tools/cmake/iree_check_test.cmake#L14-L43 somehow (I don't like that `iree_bytecode_module_for_iree_check_test_and_friends` exists in the first place)

Either `iree_bytecode_module` should learn all these tricks or it should know none of them, and `iree_static_linker_test` should use `iree_bytecode_module_for_iree_check_test_and_friends` instead :/

Reply:
There are other checks in `iree_bytecode_module_for_iree_check_test_and_friends` regarding cpu features which are not easily consolidated with the underlying `iree_bytecode_module`. At the same time, the static lib support doesn't need to be bubbled one level up.

I consolidate the platform-specific flag assignment in all files, but I would keep the  `iree_bytecode_module_for_iree_check_test_and_friends`  for now.

Reply:
SGTM. Thanks for looking!